### ISSUE TITLE
Sometimes reg_loss ==0 and it always on cpu

### ISSUE THREAD

**Issue by superMC5657** [yhenon/pytorch-retinanet#194]
Thanks for your code,I have benefited a lot
torch.stack(regressions_losses).mean(dim=0, keepdim=True)
All input tensors must be on the same device. Received cuda:0 and cpu

**Comment by EonianCoda** [comment]
I have the same problem, and I found that the problem occurred when the image has no annotations.
About some coco Images missed the annotation can see:
[https://github.com/cocodataset/cocoapi/issues/76](https://github.com/cocodataset/cocoapi/issues/76)

[/retinanet/losses.py ](https://github.com/yhenon/pytorch-retinanet/blob/master/retinanet/losses.py#L51) line 51~73 is designed for solving the problem of missing annotations, but in line 64, 78 probably have some error.

For me, just change /retinanet/losses.py 64, 78, and then the problem can be solved
from:
`regression_losses.append(torch.tensor(0).float())` 
to:
`regression_losses.append(torch.tensor(0).float().cuda())`

**Comment by wvalcke** [comment]
Problem has been reported here as well : #197 and #174
I hope @yhenon finds some time to fix it in the master branch.
Create PR ?

**Comment by superMC5657** [comment]
> I have the same problem, and I found that the problem occurred when the image has no annotations.
> About some coco Images missed the annotation can see:
> [cocodataset/cocoapi#76](https://github.com/cocodataset/cocoapi/issues/76)
> 
> [/retinanet/losses.py ](https://github.com/yhenon/pytorch-retinanet/blob/master/retinanet/losses.py#L51) line 51~78 is designed for solving the problem of missing annotations, but in line 64 probably have some error.
> 
> For me, just change /retinanet/losses.py 64 and then the problem can be solved
> from:
> `regression_losses.append(torch.tensor(0).float())`
> to:
> `regression_losses.append(torch.tensor(0).float().cuda())`
Yes, I fixed it, thank you
